docs/08_jenkins_testing_enviroment.md

```diff
@@ -0,0 +1,126 @@
+
+# API
+
+> What will the proposed API look like?
+
+The Jenkins API is described in this article: [Jenkins remote access api](https://wiki.jenkins.io/display/JENKINS/Remote+access+API).
+
+# Dependencies
+
+> What is the project blocked on?
+
+No blockers.
+
+> What will be impacted by the project?
+
+Testing of the Jenkins integration will be impacted.
+
+# Testing
+
+> How will the project be tested?
+
+This project will be tested manually.
+
+# Alternatives Considered
+
+> Summarize alternative designs (pros & cons)
+
+* Setups Jenkins server on Amazon AWS.
+    - Pros:
+         - The AWS instance is free for one year.
+         - Has an excellent [setup guide](https://d1.awsstatic.com/Projects/P5505030/aws-project_Jenkins-build-server.pdf) that explains how to create an instance and deploy the Jenkins server.
+         - Easy to maintain. You could just connect to the AWS instance and use it and use it the way that you would use a computer in front of you.
+    - Cons:
+         - You need to configure the AWS instance and firewall rules on your own, which takes a bit more time in comparison to using the Google Kubernetes deployment.
+         
+         
+* Setup Jenkins on Google Kubernetes Engine.
+    - Pros:  
+         - Easy to set up. The configuration process is fully automated, and it is needed just to configure Jenkins projects.
+           It is needed just to go through the [Jenkins Google cloud setup flow](https://console.cloud.google.com/marketplace/details/google/jenkins).
+         - Jenkins has a [free application](https://console.cloud.google.com/marketplace/details/google/jenkins) on the Google Cloud.
+    - Cons: 
+         - The Kubernetes Engine is the paid feature of the Google Cloud that uses a couple of other paid features like Compute Engine.
+         - Hard to maintain because you cannot just connect to the instance with the Jenkins server to make any changes.
+         
+         
+* Serve the Jenkins server using the local machine and ngrok.
+    - Pros:
+         - Easy to setup. You need just to run the Jenkins locally and set up the tunnel to the localhost using the ngrok.
+    - Cons: 
+         - Because Jenkins stores the data (pipeline configurations, build, build artifacts, etc.) locally,
+          using the file system, the Jenkins instance should be running on one machine each time.
+         - The server will be very human-sensitive. It means that the instance of Jenkins will be available only when it is running on someone's machine.
+         
+         
+* Setup Jenkins on Google Compute Engine.
+    - Pros:
+         - In case the Jenkins server is running on the machine, it will be needed to just connect to this machine using SSH to make any changes to the Jenkins server.
+    - Cons: 
+         - Pretty complicated setup mechanism.
+         - We should set up the Jenkins storage manually in some other place like [Cloud Storage](https://cloud.google.com/storage).
```

Comment:
Can we clarify it a bit more? If we can connect to change configuration why we need to store in some other place?